manifests/init.pp

```diff
@@ -29,14 +29,22 @@
     info('puppet_agent performs no actions on Puppet 4+')
   }
   else {
-    class { '::puppet_agent::prepare': } ->
-    class { '::puppet_agent::install': } ->
-    class { '::puppet_agent::config': } ~>
-    class { '::puppet_agent::service': }
+    if $::architecture == 'x86' and $arch == 'x64' {
+      fail('Unable to install x64 on a x86 system')
+    }
+    if $::osfamily == 'windows' {
+      class { '::puppet_agent::osfamily::windows': }
```

Comment:
Yeah, I missed that one, I thought I looked but I missed it. That one is critical yup.

For logistics though, would moving that during the run cause issues? Or are you employing the idea of copying the directories?